modules/ppcp-vaulting/src/VaultingModule.php

```diff
@@ -124,38 +72,123 @@ function( int $customer_id ) use ( $subscription_helper ) {
 			}
 		);
 
-		$asset_loader = $container->get( 'vaulting.assets.myaccount-payments' );
 		add_action(
-			'wp_enqueue_scripts',
-			function () use ( $asset_loader ) {
-				if ( is_account_page() && $this->is_payments_page() ) {
-					$asset_loader->enqueue();
-					$asset_loader->localize();
+			'woocommerce_paypal_payments_check_saved_payment',
+			function ( int $order_id, int $customer_id, string $intent ) use ( $container ) {
+				$payment_token_checker = $container->get( 'vaulting.payment-token-checker' );
+				$payment_token_checker->check_and_update( $order_id, $customer_id, $intent );
+			},
+			10,
+			3
+		);
+
+		$this->filterFailedVaultingEmailsForSubscriptionOrders( $container );
+
+		add_filter(
+			'woocommerce_payment_token_class',
+			/**
+			 * Param types removed to avoid third-party issues.
+			 *
+			 * @psalm-suppress MissingClosureParamType
+			 */
+			function ( $type ) {
+				if ( $type === 'WC_Payment_Token_PayPal' ) {
+					return PaymentTokenPayPal::class;
 				}
+
+				return $type;
 			}
 		);
 
+		add_filter(
+			'woocommerce_payment_methods_list_item',
+			/**
+			 * Param types removed to avoid third-party issues.
+			 *
+			 * @psalm-suppress MissingClosureParamType
+			 */
+			function( $item, $payment_token ) {
+				if ( ! is_array( $item ) || ! is_a( $payment_token, WC_Payment_Token::class ) ) {
+					return $item;
+				}
+
+				if ( strtolower( $payment_token->get_type() ) !== 'paypal' ) {
+					return $item;
+				}
+
+				$item['method']['brand'] = 'PayPal';
+
+				return $item;
+			},
+			10,
+			2
+		);
+
 		add_action(
-			'wc_ajax_' . DeletePaymentTokenEndpoint::ENDPOINT,
-			static function () use ( $container ) {
-				$endpoint = $container->get( 'vaulting.endpoint.delete' );
-				assert( $endpoint instanceof DeletePaymentTokenEndpoint );
+			'wp',
```

Comment:
Is this the correct hook? :thinking:

Reply:
@AlexP11223 it's the same hook used by WooCommerce here: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-form-handler.php#L34